### PR TITLE
Cryo sprite fun

### PR DESCRIPTION
**Pull request by Bawhoppen**
ss 2016-07-26 at 07 29 10

A favor for an LLA coder

### CODE REVIEW

**Comment by GunHog** [comment]
I like it! Small complaint though, he looks so squished in there. Maybe you could make the cell a little bigger?

**Comment by jughu** [comment]
Only if the eyes are closed they are technically sleeping

**Comment by Bawhoppen** [comment]
@jughu I am not messing with those overlays, this is just raptorblaze's feature anyways

**Comment by WJohn** [comment]
Neato :+1:

**Comment by RemieRichards** [comment]
@Bawhoppen close the eyes by faking it, put the overlay on the tube or something.

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -65,6 +65,60 @@
...
+		overlays.Cut()
+		return
+	if(on)
+		if(occupant)
+			icon_state = "cell-on"
+			overlays.Cut()
+			var/image/Occ = image(occupant)
+			Occ.dir = dir
+			var/icon/O = getFlatIcon(Occ)
+			O.Crop(8,1,25,32)
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: why are you cropping this?

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 2]
```diff
@@ -54,7 +54,7 @@
 	return ..()
 
 /obj/machinery/atmospherics/components/unary/cryo_cell/update_icon()
-	if(panel_open)
+/*	if(panel_open)
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: don't comment things out, remove them.

**Comment by Bawhoppen** [comment]
so much work aghhh
maybe later

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 3]
```diff
@@ -65,6 +65,60 @@
 			icon_state = "cell-on"
 	else
 		icon_state = "cell-off"
+*/
+	if(panel_open)
+		icon_state = "cell-o"
+		overlays.Cut()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```cut_overlays()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 4]
```diff
@@ -65,6 +65,60 @@
...
 	else
 		icon_state = "cell-off"
+*/
+	if(panel_open)
+		icon_state = "cell-o"
+		overlays.Cut()
+		return
+	if(state_open)
+		icon_state = "cell-open"
+		overlays.Cut()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```cut_overlays()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 5]
```diff
@@ -65,6 +65,60 @@
...
+		return
+	if(on)
+		if(occupant)
+			icon_state = "cell-on"
+			overlays.Cut()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```cut_overlays()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 6]
```diff
@@ -65,6 +65,60 @@
...
+			var/image/Occ = image(occupant)
+			Occ.dir = dir
+			var/icon/O = getFlatIcon(Occ)
+			O.Crop(8,1,25,32)
+			Occ = image(O)
+			Occ.pixel_y = 24
+			Occ.pixel_x = 7
+			Occ.layer = layer + 0.01
+			overlays += Occ
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```add_overlay()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 7]
```diff
@@ -65,6 +65,60 @@
...
+			var/image/I = new
+			I.icon = icon
+			I.icon_state = "cell-on"
+			I.layer = layer + 0.02
+			I.alpha = 190
+			overlays += I
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```add_overlay()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 8]
```diff
@@ -65,6 +65,60 @@
...
+		else
+			icon_state = "cell-on"
+			overlays.Cut()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```cut_overlays()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 9]
```diff
@@ -65,6 +65,60 @@
...
+	else
+		if(occupant)
+			icon_state = "cell-off"
+			overlays.Cut()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```cut_overlays()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 10]
```diff
@@ -65,6 +65,60 @@
...
+			var/image/Occ = image(occupant)
+			Occ.dir = dir
+			var/icon/O = getFlatIcon(Occ)
+			O.Crop(8,1,25,32)
+			Occ = image(O)
+			Occ.pixel_y = 24
+			Occ.pixel_x = 7
+			Occ.layer = layer + 0.01
+			Occ.color = rgb(120,120,120)
+			overlays += Occ
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```add_overlay()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 11]
```diff
@@ -65,6 +65,60 @@
...
+			var/image/I = new
+			I.icon = icon
+			I.icon_state = "cell-off"
+			I.layer = layer + 0.02
+			I.alpha = 190
+			overlays += I
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```add_overlay()``` please

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 12]
```diff
@@ -65,6 +65,60 @@
...
+		else
+			icon_state = "cell-off"
+			overlays.Cut()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: ```cut_overlays()``` please

**Comment by RemieRichards** [comment]
The code for this... isn't great.
Why are you making an image just to ```getFlatIcon()``` it? why not just ```getFlatIcon()``` the occupant?
You're cropping the icon, I have no idea what for since anything you're cropping should be transparent and thus not matter.

**Comment by Bawhoppen** [comment]
I literally said I just copypasted his code in a hurry cause I kept forgetting about it, I intended to clean this up I just needed to get the image for it so I could make the PR

**Comment by RemieRichards** [comment]
@Bawhoppen actually you said nothing of the sort, you just said this was somebody else's feature, which to me read as you doing it FOR them, especially coupled with the op about this being a favour.

**Comment by Bawhoppen** [comment]
okay well it doesn't matter either way I will deal with it

**Comment by Fox-McCloud** [comment]
we have sprites for this on Para, though you'd have to modify it for the open/closed machine states.

**Comment by WJohn** [comment]
I kind of like our iconic cryo tube sprites, we've had them for so long, but I'd be lying if I said those weren't really nice, too.

**Comment by Shapsy** [comment]
Dunno, I think its tense when someone is in cryo in medbay and someone hunting for them runs through but doesn't check cryo, it'd be a shame if you couldn't hide in there anymore

**Comment by PKPenguin321** [comment]
yeah i was about to say the same thing, this kills hiding in cryo tubes

**Comment by Core0verload** [comment]
Don't like it, it simply looks ugly when a mob is inside.

**Comment by Bawhoppen** [comment]
only tg would complain about tactical cryo

**Comment by PKPenguin321** [comment]
that is not true, hiding in a cryo tube as the blood covered serial murderer stalks past while you stay as quiet as you can is immersive as fuck

**Comment by anconfuzedrock** [comment]
how does this work with lizards

**Comment by Bawhoppen** [comment]
okay before I fix the code do people want this or not cause like if its not going to be merged I'm not bothering

**Comment by optimumtact** [comment]
If you ask me para's sprite looks better

**Comment by Core0verload** [comment]
It's bottom is just ugly as fuck.

**Comment by optimumtact** [comment]
Please make the request code changes.

**Comment by Bawhoppen** [comment]
I will but I'm not sure if I even want this PR anymore, oh well we can always revert it I guess

**Comment by GunHog** [comment]
It is a good change, Baw, just work out the kinks!

**Comment by Core0verload** [comment]
No, it's a bad change, mobs in cryotubes look ugly as fuck.

**Comment by KorPhaeron** [comment]
stale

**Comment by Bawhoppen** [comment]
wait you closed this

**Comment by Bawhoppen** [comment]
I had just come here to close it and say I'd return to it

**Pull request closed without merging**